### ISSUE TITLE
Problems with crunchy postgres on nerc-ocp-test

### ISSUE THREAD

**Issue by larsks** [nerc-project/operations#477]
We are trying to deploy a PostgresCluster on nerc-ocp-test using the crunchy postgres operator, but the database pods are failing with:

```
Initializing ...
::postgres-operator: uid::26
::postgres-operator: gid::26
::postgres-operator: postgres path::/usr/pgsql-15/bin/postgres
::postgres-operator: postgres version::postgres (PostgreSQL) 15.4
::postgres-operator: config directory::/pgdata/pg15
::postgres-operator: data directory::/pgdata/pg15
install: cannot change permissions of ‘/pgdata/pg15’: No such file or directory
stat: cannot statx '/pgdata/pg15': No such file or directory
```

The error `cannot change permissions of /pgdata/pg15` is cropping up because the `postgres-data` volume is getting mounted on `/pgdata` with `root:root` permissions. The postgres user (uid 26) is not ablet to write to this directory.

**Comment by larsks** [comment]
The solution appears to be setting `spec.openshift: false` in the PostgresCluster manifest. This causes the operator to add `fsGroup: 26` to `spec.securityContext` for the database pods, which causes the mountpoint to have appropriate permissions.

Setting `openshift: false` does not appear to cause any other changes in the pod manifest.

I have posted [this question](https://groups.google.com/a/crunchydata.com/g/postgres-operator/c/PSmmhqWDlr0) to the postgres-operator mailing list.

**Comment by computate** [comment]
@larsks @dystewart that surprises me, because I've always had to configure `openshift: true` on OpenShift and `openshift: false` on Kubernetes. I have a full example of a working Crunchy Postgres cluster here on OpenShift prod: 

```bash
$ oc -n smart-village-faeeb6c get postgrescluster/postgres -o yaml | grep 'openshift:'
  openshift: true
```

I checked the permissions of the /pgdata and /pgdata/pg15 directories below: 

```bash
sh-4.4$ ls -ldZ /pgdata/
drwxrwsr-x. 6 root postgres system_u:object_r:container_file_t:s0:c20,c31 4096 Jan 16 19:22 /pgdata/
sh-4.4$ ls -ldZ /pgdata/
lost+found/ pg15/       pg15_wal/   pgbackrest/ 
sh-4.4$ ls -ldZ /pgdata/pg15/
drwx--S---. 19 postgres postgres system_u:object_r:container_file_t:s0:c20,c31 4096 Mar 12 14:43 /pgdata/pg15/
```

**Comment by larsks** [comment]
@computate it surprises me, too, because we have successfully used the postgres operator elsewhere! And yet, this is the behavior I observed on the test cluster.

**Comment by larsks** [comment]
The postgres pod for the smartvillage project, for which the `postgrescluster` resource has `openshift: true`, has `runAsUser: 1000970000` in both `spec.securityContext` and `spec.containers.*.securityContext`.

This is what allows it to work, *and* this is what is missing in the pod on the test cluster when setting `openshift: true` (or allowing it to be autodetected).

**Comment by computate** [comment]
@larsks I'm curious how you've set up your instance and dataVolumeClaimSpec, for example: 

```yaml
  instances:
    - dataVolumeClaimSpec:
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 20Gi
```

**Comment by larsks** [comment]
The prod on test clusters have different versions of the postgres operator.

- on nerc-ocp-prod: postgresoperator.v5.5.0
- on nerc-ocp-test: postgresoperator.v5.4.3

Maybe that's the source of the problem? We should upgrade the operator on the test cluster (and possibly figure out why it hasn't updated automatically).

**Comment by larsks** [comment]
> I'm curious how you've set up your instance and dataVolumeClaimSpec, for example:

I don't think the dataVolumeClaimSpec is going to be relevant (it controls the pvc, but doesn't have any impact on the pod configuration). In any case, the dataVolumeClaimSpec on the test cluster is pretty much identical.

**Comment by computate** [comment]
@larsks yeah, it probably is worth upgrading to 5.5.0 on the test cluster.

**Comment by larsks** [comment]
This seems to have been resolved by the upgraded to OpenShift 4.13.13 (and the concomitant upgrade of the postgres operator to v5.5.1).
